### PR TITLE
net: fix UDP raw sockets

### PR DESCRIPTION
**Pull request by arbrauns**
With `AF_INET`, `SOCK_RAW`, `IPPROTO_UDP`, the network stack should not insert its own UDP header; see `raw(7)`.

I'm not sure if this is the best way to fix this, but it solves my specific problem.

### CODE REVIEW

**Comment by arbrauns** [comment]
[`net_context_check()`](https://github.com/zephyrproject-rtos/zephyr/blob/43fc0f5184d0ec593e83055e7518254dfbe1e6ca/subsys/net/ip/net_context.c#L188-L189) explicitly accepts them, which I interpreted as "claiming support".

**Comment by rlubos** [comment]
Looks like a regression introduced in [net_context_get() refactor](https://github.com/zephyrproject-rtos/zephyr/commit/c57650c4033e8bc36c30fe2e30980b94f6d50026) - before, there was an explicit check that SOCK_RAW is only used with AF_PACKET or AF_CAN.

File: subsys/net/ip/net_context.c [hunk 1]
```diff
@@ -1357,12 +1357,14 @@ static int context_setup_udp_packet(struct net_context *context,
...
-				     &context->local)->sin_port,
-			     dst_port);
-	if (ret) {
-		return ret;
+	if (net_context_get_type(context) != SOCK_RAW) {
+		ret = net_udp_create(pkt,
+				     net_sin((struct sockaddr *)
+					     &context->local)->sin_port,
+				     dst_port);
+		if (ret) {
```
Comment (tbursztyka) [inline subsys/net/ip/net_context.c:1365]: remove this test

File: subsys/net/ip/net_context.c [hunk 2]
```diff
@@ -1357,12 +1357,14 @@ static int context_setup_udp_packet(struct net_context *context,
...
+			return ret;
+		}
 	}
 
 	ret = context_write_data(pkt, buf, len, msg);
```
Comment (tbursztyka) [inline subsys/net/ip/net_context.c:1370]: put this on a else after previous if()

Comment (arbrauns) [reply subsys/net/ip/net_context.c:1370]: Sorry, I don't understand how you want this to be ordered, could you post the code you'd like to see?

Comment (jukkar) [reply subsys/net/ip/net_context.c:1370]: I believe @tbursztyka meant something like this. As there are now multiple `if(ret) { ... }` checks in this function, you could remove one of them. So the code could look like this:
```
	if (net_context_get_type(context) != SOCK_RAW) {
		ret = net_udp_create(pkt,
				     net_sin((struct sockaddr *)
					     &context->local)->sin_port,
				     dst_port);
	}

	if (ret == 0) {
		ret = context_write_data(pkt, buf, len, msg);
	}

	return ret;
}
```

Comment (arbrauns) [reply subsys/net/ip/net_context.c]: Alright, personally I think that just makes the diff noisier, but I've done it.

**Pull request closed without merging**